appcenter-data/android/src/main/java/com/microsoft/appcenter/reactnative/data/AppCenterReactNativeDataModule.java

```diff
@@ -1,32 +1,135 @@
 
-public class AppCenterReactNativeDataModule extends ReactContextBaseJavaModule {
+import com.microsoft.appcenter.AppCenter;
+import com.microsoft.appcenter.data.Data;
+import com.microsoft.appcenter.data.models.DocumentWrapper;
+import com.microsoft.appcenter.reactnative.shared.AppCenterReactNativeShared;
+import com.microsoft.appcenter.utils.async.AppCenterConsumer;
 
-  private static final String DURATION_SHORT_KEY = "SHORT";
-  private static final String DURATION_LONG_KEY = "LONG";
 
-  public AppCenterReactNativeDataModule(ReactApplicationContext reactContext) {
-    super(reactContext);
-  }
+public class AppCenterReactNativeDataModule extends BaseJavaModule {
 
-  @Override
-  public String getName() {
-    return "AppCenterReactNativeData";
-  }
+    public AppCenterReactNativeDataModule(Application application, boolean startEnabled) {
+        AppCenterReactNativeShared.configureAppCenter(application);
 
-  @ReactMethod
-  public void show(String message, int duration) {
-    Toast.makeText(getReactApplicationContext(), message, duration).show();
-  }
+        if (AppCenter.isConfigured()) {
+            AppCenter.start(Data.class);
+
+            if (!startEnabled) {
+                Data.setEnabled(false);
+            }
+        }
+    }
+
+    @Override
+    public String getName() {
+        return "AppCenterReactNativeData";
+    }
+
+    @ReactMethod
+    public void read(String documentId, String partition, final Promise promise) {
+        Data.read(documentId, JsonElement.class, partition).thenAccept(new AppCenterConsumer<DocumentWrapper<JsonElement>>() {
+            @Override
+            public void accept(DocumentWrapper<JsonElement> documentWrapper) {
+                JsonElement element = documentWrapper.getDeserializedValue();
+
+                if (element.isJsonPrimitive()) {
+                    JsonPrimitive jsonPrimitive = element.getAsJsonPrimitive();
+
+                    if (jsonPrimitive.isString()) {
+                        promise.resolve(jsonPrimitive.getAsString());
+                    } else if (jsonPrimitive.isNumber()) {
+                        promise.resolve(jsonPrimitive.getAsNumber());
+                    } else if (jsonPrimitive.isBoolean()) {
+                        promise.resolve(jsonPrimitive.getAsBoolean());
+                    }
+                } else if (element.isJsonObject()) {
+                    JsonObject jsonObject = element.getAsJsonObject();
+                    WritableMap writableMap = convertJsonObjectToWritableMap(jsonObject);
+
+                    promise.resolve(writableMap);
+                } else if (element.isJsonArray()) {
+                    JsonArray jsonArray = element.getAsJsonArray();
+                    WritableArray writableArray = convertJsonArrayToWritableArray(jsonArray);
+
+                    promise.resolve(writableArray);
+                } else {
+                    promise.resolve(element.getAsJsonNull());
```

Comment:
No we need to return simply null here to translate to null in Javascript. JsonNull is an object that will not be convertible directly.

Reply:
Done